### ISSUE TITLE
ncdump code breaks HDF5-1.8 builds

### ISSUE THREAD

**Issue by edwardhartnett** [Unidata/netcdf-c#1776]
Jenkins is showing me this build error on HDF5-1.8 builds. 

```
gcc -DHAVE_CONFIG_H -I. -I../../../ncgen -I..  -I../../../include -I../../../oc2 -I/usr/local/hdf5-1.8.21_szip/include   -MT ncdumpchunks.o -MD -MP -MF $depbase.Tpo -c -o ncdumpchunks.o ../../../ncgen/ncdumpchunks.c &&\
mv -f $depbase.Tpo $depbase.Po
../../../ncgen/ncdumpchunks.c: In function ‘main’:
../../../ncgen/ncdumpchunks.c:108:12: warning: implicit declaration of function ‘H5Dread_chunk’; did you mean ‘H5Pget_chunk’? [-Wimplicit-function-declaration]
  108 |         if(H5Dread_chunk(datasetid, dxpl_id, offset, &filter_mask, chunkdata) < 0) abort();
      |            ^~~~~~~~~~~~~
      |            H5Pget_chunk
```

**Comment by edwardhartnett** [comment]
@WardF I suggest you add a HDF5-1.8 build to Travis. Lots of users use 1.8 because it's faster...

**Comment by DennisHeimbigner** [comment]
Apparently, the history is this:
H5DO_read_chunk was introduced in 1.10.2 and 1.8.19
H5DO_read_chunk was renamed H5Dread_chunk in 1.10.3

The ncdumpchunks.c program was added to the nczarr code as a test
program to verify that I was getting chunking to match that of HDF5.
So that program needs to be conditional on 1.10.3 or later,
which is ok with me. I will try to add that constraint.

**Comment by edhartnett** [comment]
I figured it was something like this.

Actually, this is the very function I am using on configure.ac to ensure
that parallel compression is available:

   # Check to see if HDF5 library is 1.10.3 or greater. If so, allows
   # parallel I/O with filters. This allows zlib/szip compression to
   # be used with parallel I/O, which is very helpful to HPC users.
   if test "x$ac_cv_func_H5Dread_chunk" = xyes; then
      AC_DEFINE([HDF5_SUPPORTS_PAR_FILTERS], [1], [if true, HDF5 is at
least version 1.10.3 and allows parallel I/O with zip])
      hdf5_supports_par_filters=yes
   fi
   AC_MSG_CHECKING([whether HDF5 allows parallel filters])
   AC_MSG_RESULT([$ac_cv_func_H5Dread_chunk])

**Comment by WardF** [comment]
Thanks @DennisHeimbigner, @edhartnett.  I'm parameterizing the docker containers we use, so that we can specify the HDF5 version to use.  This should let us at least spot-test on travis.  Dennis, if you'd like me to add the fence-posting on this, let me know; I don't mind doing it, but also don't want to accidentally duplicate effort if you've already started.

**Comment by edwardhartnett** [comment]
It would be good to get a fix for this soon. Support for HDF5 1.8.* has been painstakingly maintained all this time.

Since this is causing my CI to fail on 1.8 builds, how can we tell is changes are breaking it further?

**Comment by edwardhartnett** [comment]
This was fixed by PR #1790 
Thanks @gsjaardema !
